src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,82 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
-                            busyResponse.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
-                            busyResponse.delayedActionTime.SetValue(response.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
-                            handler->AddResponse(cachedCommandPath, busyResponse);
-                            handle.Release();
-                            return;
-                        }
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
-                    auto targetNodeId
-                        = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+                      CommandHandler * handler = EnsureValidState(handle, cachedCommandPath, "QueryImage", data, error);
+                      VerifyOrReturn(handler != nullptr);
 
-                    char uriBuffer[kMaxBDXURILen];
-                    MutableCharSpan uri(uriBuffer);
-                    err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
+                      ChipLogDetail(Controller, "QueryImage: application responded with: %s",
+                          [[data description] cStringUsingEncoding:NSUTF8StringEncoding]);
 
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
-                    handle.Release();
-                    return;
-                }
-                if (!hasUpdate) {
-                    // Send whatever error response our delegate decided on.
-                    handler->AddResponse(cachedCommandPath, response);
-                } else {
-                    // We must have isBDXProtocolSupported false.  Send the corresponding error status.
-                    Commands::QueryImageResponse::Type protocolNotSupportedResponse;
-                    protocolNotSupportedResponse.status
-                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
-                    handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
-                }
-                handle.Release();
-                gOtaSender.ResetState();
-            }
+                      Commands::QueryImageResponse::Type response;
+
+                      // If BDX protocol is not supported, send a response with status - Protocol Not supported
```

Comment:
I think the logic here needs to be:
1. If delegate responded with any status other than MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable, respond with that status _and whatever other information the delegate put in the response_, done.
2. If other side does not support MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous, respond with ProtocolNotSupported.
3. Try to respond with success.

Reply:
done. kept the protocol supported at the top. will change based on offline discussion if needed

Reply:
added a comment as to why we followed the sequence of protocol check first, then update image not available and image available